This is not catching the packaging regression. I suspect it is because we are not getting suitable environment isolation.

The errors [encountered in the ocaml-ci](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/cf61b44b20bb45b17951d5b49adb72c2ca63771a/variant/distributions,ubuntu-22.04-ocaml-4.14,dune-rpc.3.24.0~alpha0) are

```
dune-rpc.3.24.0~alpha0 is not installed. Install it? [Y/n] y
The following actions will be performed:
=== install 11 packages
  - install csexp       1.5.2                  [required by dune-rpc]
  - install dune        3.24.0~alpha0          [required by dune-rpc]
  - install dune-rpc    3.24.0~alpha0 (pinned)
  - install dyn         3.24.0~alpha0          [required by dune-rpc]
  - install fs-io       3.24.0~alpha0          [required by stdune]
  - install ocamlc-loc  3.24.0~alpha0          [required by dune-rpc]
  - install ordering    3.24.0~alpha0          [required by dune-rpc]
  - install pp          2.0.0                  [required by dune-rpc]
  - install stdune      3.24.0~alpha0          [required by dune-rpc]
  - install top-closure 3.24.0~alpha0          [required by stdune]
  - install xdg         3.24.0~alpha0          [required by dune-rpc]

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
-> retrieved csexp.1.5.2  (cached)
-> retrieved dune.3.24.0~alpha0, dune-rpc.3.24.0~alpha0, dyn.3.24.0~alpha0, fs-io.3.24.0~alpha0, ocamlc-loc.3.24.0~alpha0, ordering.3.24.0~alpha0, stdune.3.24.0~alpha0, top-closure.3.24.0~alpha0, xdg.3.24.0~alpha0  (cached)
-> retrieved pp.2.0.0  (cached)
-> installed dune.3.24.0~alpha0
-> installed csexp.1.5.2
-> installed fs-io.3.24.0~alpha0
-> installed ordering.3.24.0~alpha0
-> installed pp.2.0.0
-> installed top-closure.3.24.0~alpha0
-> installed xdg.3.24.0~alpha0
-> installed dyn.3.24.0~alpha0
-> installed ocamlc-loc.3.24.0~alpha0
[ERROR] The compilation of stdune.3.24.0~alpha0 failed at "dune build -p stdune -j 255 @install".

#=== ERROR while compiling stdune.3.24.0~alpha0 ===============================#
# context              2.5.1 | linux/x86_64 | ocaml-base-compiler.4.14.3 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/stdune.3.24.0~alpha0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p stdune -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/stdune-7-b7c611.env
# output-file          ~/.opam/log/stdune-7-b7c611.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w @1..3@5..28@31..39@43@46..47@49..57@61..62@67@69-40 -strict-sequence -strict-formats -alert -unstable -g -bin-annot -I otherlibs/stdune/src/.stdune.objs/byte -I /home/opam/.opam/4.14/lib/csexp -I /home/opam/.opam/4.14/lib/dyn -I /home/opam/.opam/4.14/lib/fs-io -I /home/opam/.opam/4.14/lib/ordering -I /home/opam/.opam/4.14/lib/pp -I /home/opam/.opam/4.14/lib/top-closure -no-alias-deps -open Stdune__ -o otherlibs/stdune/src/.stdune.objs/byte/stdune__Mutex0.cmo -c -impl otherlibs/stdune/src/mutex0.ml)
# File "otherlibs/stdune/src/mutex0.ml", line 1, characters 5-10:
# 1 | open Mutex
#          ^^^^^
# Error: Unbound module Mutex
# Hint: Did you mean Mutex0?
```

Whereas the current CI job solves with

```
<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
[dune-rpc.dev] synchronised (no changes)

The following actions will be performed:
=== install 11 packages
  ∗ csexp       1.5.2        [required by dune-rpc]
  ∗ dune        dev (pinned) [required by dune-rpc]
  ∗ dune-rpc    dev (pinned)
  ∗ dyn         dev (pinned) [required by dune-rpc]
  ∗ fs-io       dev (pinned) [required by stdune]
  ∗ ocamlc-loc  dev (pinned) [required by dune-rpc]
  ∗ ordering    dev (pinned) [required by dune-rpc]
  ∗ pp          2.0.0        [required by dune-rpc]
  ∗ stdune      dev (pinned) [required by dune-rpc]
  ∗ top-closure dev (pinned) [required by stdune]
  ∗ xdg         dev (pinned) [required by dune-rpc]

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved csexp.1.5.2  (cached)
⬇ retrieved dune-rpc.dev  (no changes)
⬇ retrieved dune.dev  (no changes)
⬇ retrieved dyn.dev  (no changes)
⬇ retrieved fs-io.dev  (no changes)
⬇ retrieved pp.2.0.0  (cached)
⬇ retrieved ordering.dev  (no changes)
⬇ retrieved stdune.dev  (no changes)
⬇ retrieved top-closure.dev  (no changes)
⬇ retrieved ocamlc-loc.dev  (no changes)
⬇ retrieved xdg.dev  (no changes)
∗ installed dune.dev
∗ installed csexp.1.5.2
∗ installed pp.2.0.0
∗ installed fs-io.dev
∗ installed ordering.dev
∗ installed top-closure.dev
∗ installed xdg.dev
∗ installed dyn.dev
∗ installed ocamlc-loc.dev
∗ installed stdune.dev
∗ installed dune-rpc.dev
Done.
```

Tomorrow I'll pick up here to try to figure out what we are missing in our tests.

--- a/.github/workflows/isolated-package-build.yml
+++ b/.github/workflows/isolated-package-build.yml
@@ -18,6 +18,12 @@ on:
 jobs:
   packaging-check:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ocaml-compiler:
+          - "4.14" # The oldest supported compiler version
+          - "5"    # The latest available compiler version
 
     steps:
       - uses: actions/checkout@v6
@@ -38,7 +44,7 @@ jobs:
 
       - uses: ocaml/setup-ocaml@v3
         with:
-          ocaml-compiler: 5.4
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
           dune-cache: true
           opam-pin: false
 

